### DISCUSSION TITLE
working with a file.

### DISCUSSION THREAD

**Discussion by ihsinme** [strongswan/strongswan#740] [General]
Good afternoon.
Do I understand correctly that these files are opened without restricting access rights (0666).

https://github.com/strongswan/strongswan/blob/f4bfdec21f3400d8de3872cad3b8f00c3499471b/src/libcharon/plugins/save_keys/save_keys_listener.c#L302

https://github.com/strongswan/strongswan/blob/f4bfdec21f3400d8de3872cad3b8f00c3499471b/src/libcharon/plugins/save_keys/save_keys_listener.c#L356

it might be worth limiting them, at least to 0644.

thanks for taking the time to answer

**Comment by tobiasbrunner** [comment]
I guess we could change that. But since this plugin is not intended to be used on productive systems, only for development and debugging, I don't really see a problem with this.

**Reply by Gor1234567** [reply]
Is ikev2_decryption_table file limited (how many lines/connections could it contain)?

Thanks, Tobias!

**Reply by tobiasbrunner** [reply]
No idea. You have to ask the Wireshark developers about that. But it's really just a development/debugging tool. If you have a lot of lines in there, you are doing something wrong.
